swap_meet/vendor.py

```diff
@@ -1,2 +1,88 @@
+from unicodedata import category
+from swap_meet.item import Item
+
 class Vendor:
-    pass
\ No newline at end of file
+    
+    def __init__(self, inventory = None):
+        if not inventory:
+            self.inventory = []
+        else:
+            self.inventory = inventory 
+            
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+
+
+    def get_by_category(self, category):
+        if not category:
+            return False
+        category_list = []
+        for item in self.inventory:
+            if item.category == category:
+                category_list.append(item)
+        return category_list 
+
+
+    def swap_items(self, friend, my_item, their_item):
+        if my_item not in self.inventory:
+            return False
+        if their_item not in friend.inventory:
+            return False
+
+        self.inventory.remove(my_item)
+        friend.inventory.append(my_item) 
+        friend.inventory.remove(their_item)
+        self.inventory.append(their_item)
+
+        if their_item not in self.inventory:
+            return False
+        else:
+            return True
+
+
+    def swap_first_item(self, friend):
+        if self.inventory == []:
+            return False
+        else: 
+            my_item = self.inventory[0]
+        if friend.inventory == []:
+            return False
+        else:
+            their_item = friend.inventory[0]
```

Comment:
If `friend.inventory` is empty, we'll create the `my_item` variable even though it will never be used. I recommend using an if to validate the input first before creating variables:
```suggestion
        if not self.inventory or not friend.inventory:
            return False

        my_item = self.inventory[0]
        their_item = friend.inventory[0]
```